Dao ui improvements

ManfredKarrer opened the pull request.

ManfredKarrer committed:
Fix/add comments

ManfredKarrer committed:
Add ButtonCell and PromptText to ComboBox in FormBuilder

Default ComboBox does not show promptText after clear selection.
https://stackoverflow.com/questions/50569330/how-to-reset-combobox-and-display-prompttext?noredirect=1&lq=1

ManfredKarrer committed:
Improve setFitToRowsForTableView method

ManfredKarrer committed:
Add new items to Param

- Add COMPENSATION_REQUEST_MIN_AMOUNT and COMPENSATION_REQUEST_MAX_AMOUNT
As BSQ based validation values can change over time if BSQ value
rise we need to support that in the Params as well.

- Add validateParamValue to ChangeParamValidator

ManfredKarrer committed:
Improve param change selection (WIP)

- Show current formatted value
- Parse input to correct domain value
- Add validation (WIP)
- Add boundary checks (WIP)

ManfredKarrer committed:
Add Generic- and removeAsset proposals

ManfredKarrer committed:
Remove asset from comboBoxes if it was removed by voting

ManfredKarrer committed:
Add checkTradeActivity method, remove ConvertToTradeStatistics2

ManfredKarrer committed:
Merge branch 'master' into Dao-UI-improvements

sqrrm commented on core/src/main/java/bisq/core/dao/governance/proposal/compensation/CompensationConsensus.java:
```diff
@@ -18,18 +18,21 @@
 public class CompensationConsensus extends ProposalConsensus {
-    public static Coin getMinCompensationRequestAmount() {
-        return Coin.valueOf(1_000); // 10 BSQ
+    public static Coin getMinCompensationRequestAmount(BsqStateService bsqStateService, int chainHeadHeight) {
```
Why call the argument chainHeadHeight instead of just chainHeight?

ManfredKarrer replied:
Yes should be chainHeight, was mixing it up a bit. will change...

ManfredKarrer committed:
Merge branch 'master' into Dao-UI-improvements

sqrrm commented on core/src/main/java/bisq/core/dao/governance/proposal/param/ChangeParamValidator.java:
```diff
@@ -41,9 +47,109 @@ public void validateDataFields(Proposal proposal) throws ValidationException {
+                break;
+            case BLIND_VOTE_FEE:
+                break;
+            case COMPENSATION_REQUEST_MIN_AMOUNT:
```
There should also be a check that `COMPENSATION_REQUEST_MIN_AMOUNT < COMPENSATION_REQUEST_MAX_AMOUNT`, both current and proposed.

ManfredKarrer replied:
Yes the validations are very much WIP...

sqrrm commented on core/src/main/java/bisq/core/dao/governance/proposal/param/ChangeParamValidator.java:
```diff
@@ -41,9 +47,109 @@ public void validateDataFields(Proposal proposal) throws ValidationException {
+    @VisibleForTesting
+    long getNewValueByPercentChange(Param param, long percentChange) {
+        checkArgument(percentChange > -100, "percentChange must be bigger than -100");
+        return (getCurrentValue(param) * 100 * (100 + percentChange)) / 100;
```
This multiplies by 100 one time too many.

sqrrm commented on core/src/main/java/bisq/core/util/BsqFormatter.java:
```diff
@@ -114,4 +116,101 @@ public Coin parseSatoshiToBtc(String satoshi) {
+            case BSQ_TAKER_FEE_IN_PERCENT:
+            case BTC_MAKER_FEE_IN_PERCENT:
+            case BTC_TAKER_FEE_IN_PERCENT:
+                return formatToPercentWithSymbol(value / 10000d);
```
Why is the value divided by 10000 rather than 100 to get percent?

ManfredKarrer replied:
Because we want to support 2 decimal places (e.g. 12.34%) so its 100 * 100 = 10 000

sqrrm replied:
Ah, now I get it. Probably good to add a comment about that to make it clear

sqrrm reviewed (commented):
I would like some clarification on my comments on how calculations are done. Something seems strange about it, either it's correct and some code comments are needed or something isn't working as intended.

ManfredKarrer committed:
Merge branch 'Dao-UI-improvements'

ManfredKarrer committed:
Rename chainHeadHeight to chainHeight

ManfredKarrer commented:
The validation code for params is WIP and I am not sure if the current implementation is intuitive (range set by % of current value...). I will keep that for a bit later and try to get other high prio stuff done first. But feel free to work on that if you want.

ManfredKarrer committed:
Persist EvaluatedProposalList

ManfredKarrer committed:
Persist DecryptedBallotsWithMeritsList

ManfredKarrer committed:
Add handling for missing blindVote data

ManfredKarrer committed:
Add handling for missing proposal data

ManfredKarrer committed:
Add VoteResultExceptionHandler

ManfredKarrer committed:
Merge remote-tracking branch 'upstream/master'

ManfredKarrer committed:
Merge branch 'master' into Dao-UI-improvements

ManfredKarrer committed:
Remove PHASE_BREAK4, adjust phase periods and fees

ManfredKarrer committed:
Reduce default btc miner fee from 200 to 50 sat/byte

ManfredKarrer committed:
Add comments, cleanups

ManfredKarrer committed:
Remove other base currencies from FeeService

ManfredKarrer committed:
Adjust comments

ManfredKarrer committed:
Rename fields and params

ManfredKarrer committed:
Rename support for other base currencies in getDefaultMinFeePerByte

ManfredKarrer committed:
Make ProposalConsensus and CompensationConsensus methods static

ManfredKarrer committed:
Support trade fees defined DAO params

- Use DAO Params to get the trade fees
- Rename existing params
- Add min fee values

ManfredKarrer committed:
Adjust DAO params

ManfredKarrer committed:
Update comments

ManfredKarrer committed:
Change log level for printing asset trade stats

sqrrm commented on core/src/main/java/bisq/core/dao/node/full/network/FullNodeNetworkService.java:
```diff
@@ -173,6 +189,24 @@ public void onFault(String errorMessage, @Nullable Connection connection) {
+            ObservableList<BlindVotePayload> appendOnlyStoreList = blindVoteListService.getAppendOnlyStoreList();
+            appendOnlyStoreList
+                    .forEach(blindVotePayload -> {
+                        // We want a random delay between 0.1 and 30 sec. depending on the number of items
```
What's the reason to depend on the number of items?

ManfredKarrer replied:
We don't want to stress the network with many broadcasts, so if there are many objects it will be delayed, the more objects the longer the dealy, but not more then 30 sec.

sqrrm commented on core/src/main/java/bisq/core/provider/fee/FeeService.java:
```diff
@@ -51,42 +54,72 @@
-    // 2 BSQ or 200 BSQ-satoshi. About 2 USD if 1 BSQ = 1 USD for a 1 BTC trade which is about 10% of a normal BTC fee.
-    private static final long DEFAULT_TAKER_FEE_BSQ = 200;
-    private static final long DEFAULT_MAKER_FEE_BSQ = 200;
+    public static Coin getMakerFeePerBtc(boolean currencyForFeeIsBtc) {
```
This is a bit confusing. Probably better to have separate methods for the two coins. If necessary to have just one method it should be clear what the Coin returned might be.

ManfredKarrer replied:
Hm.. the handling of the BTC or BSQ fee is not very nice atm... would require a bit more refactoring at the clients side... Atm I don't want to touch more as it is needed so prefer to keep it but yes, its an open TODO...

sqrrm reviewed (approved):
utACK

Some comments on parts that might need more documentation but please go ahead and merge.
